### PR TITLE
Use lowest included slot for precompute

### PR DESCRIPTION
**Pull request by terencechain**
When we precompute for process epoch, as part of validator record we should save the lowest inclusion slot for attestation

### CODE REVIEW

File: beacon-chain/core/epoch/precompute/reward_penalty_test.go [hunk 1]
```diff
@@ -55,15 +55,15 @@ func TestProcessRewardsAndPenaltiesPrecompute(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Indices that voted
-	wanted := uint64(32000001776)
+	// Indices that voted everything except for head, lost a bit money
+	wanted := uint64(31999995452)
```
Comment (shayzluf) [inline beacon-chain/core/epoch/precompute/reward_penalty_test.go]: can we calculate this number easily?

File: beacon-chain/core/epoch/precompute/reward_penalty_test.go [hunk 2]
```diff
@@ -55,15 +55,15 @@ func TestProcessRewardsAndPenaltiesPrecompute(t *testing.T) {
...
 	if state.Balances[4] != wanted {
 		t.Errorf("wanted balance: %d, got: %d",
 			wanted, state.Balances[4])
 	}
 
-	// Indices that did not vote - lose money
-	wanted = uint64(31999797616)
+	// Indices that did not vote, lost more money
+	wanted = uint64(31999949392)
```
Comment (shayzluf) [inline beacon-chain/core/epoch/precompute/reward_penalty_test.go]: what about a case for indices that got rewarded by voting on everything?

Comment (terencechain) [reply beacon-chain/core/epoch/precompute/reward_penalty_test.go]: Updated the test case to use minimal config. Now a validator correctly votes on everything and makes money :)

File: beacon-chain/core/epoch/precompute/reward_penalty_test.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/epoch/precompute/reward_penalty_test.go]: It's pretty tedious, i think what we have is sufficient enough for unit test, the rest is covered as part of spec test

Comment (shayzluf) [reply beacon-chain/core/epoch/precompute/reward_penalty_test.go]: sg. thanks

**Pull request merged**